frontend/packages/topology/src/components/workload/vpa-tab-section.tsx

```diff
@@ -51,11 +51,14 @@ export const useVpaSideBarTabSection: DetailsTabSectionExtensionHook = (element:
   }
 
   const resource = getResource(element);
+  const { namespace } = resource.metadata;
+  const vpasForNamespace = vpas.filter((vpa) => vpa.metadata.namespace === namespace);
+
   const verticalPodAutoscaler = getVerticalPodAutoscalerForResource(vpas, resource);
 
   const section = verticalPodAutoscaler ? (
     <TopologySideBarTabSection>
-      <VPATabSection vpas={vpas} />
+      <VPATabSection vpas={vpasForNamespace} />
```

Comment:
```suggestion
      <VPATabSection vpas={verticalPodAutoscaler} />
```
is what you want, no?  On the Deployment details page, we assume there is only one VPA.  I would think we want the same here.  If not, then I think we need to revisit the Deployment details page as well.